tests/test_mail.py

```diff
@@ -0,0 +1,164 @@
+
+client = TestClient(app)
+
+
+@pytest.fixture
+def configured_smtpd(smtpd):
+    """
+    Overrides the SMTP related configuration to use a mock SMTP server
+    :param smtpd: the smtpdfix fixture that represents an SMTP server
+    :return: smtpd
+    """
+
+    def override_settings():
+        return config.Settings(smtp_server=smtpd.hostname,
+                               smtp_port=smtpd.port)
+
+    app.dependency_overrides[config.get_settings] = override_settings
+    yield smtpd
+    app.dependency_overrides.pop(config.get_settings)
+
+
+def test_read_main():
+    response = client.get("/")
+    assert response.status_code == 200
+
+
+def test_send_mail_no_body(configured_smtpd):
+    response = client.post("/mail/invitation/")
+    assert response.status_code == 422
+    assert response.json() == {'detail': [{
+        'loc': ['body'],
+        'msg': 'field required',
+        'type': 'value_error.missing'}]}
+    assert len(configured_smtpd.messages) == 0
+
+
+@pytest.mark.parametrize("body,expected_json", [
+    (
+            {"sender_name": "string", "recipient_name": "string"},
+            {'detail': [{
+                'loc': ['body', 'recipient_mail'],
+                'msg': 'field required',
+                'type': 'value_error.missing'}]},
+    ),
+
+    (
+            {"sender_name": "string", "recipient_mail": "test@mail.com"},
+            {'detail': [{
+                'loc': ['body', 'recipient_name'],
+                'msg': 'field required',
+                'type': 'value_error.missing'}]},
+    ),
+    (
+            {"recipient_name": "string", "recipient_mail": "test@mail.com"},
+            {'detail': [{
+                'loc': ['body', 'sender_name'],
+                'msg': 'field required',
+                'type': 'value_error.missing'}]},
+    ),
+    (
+            {"sender_name": "string"},
+            {'detail': [
+                {'loc': ['body', 'recipient_name'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'},
+                {'loc': ['body', 'recipient_mail'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'}
+            ]}
+    ),
+    (
+            {"recipient_name": "string"},
+            {'detail': [
+                {'loc': ['body', 'sender_name'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'},
+                {'loc': ['body', 'recipient_mail'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'}
+            ]}
+    ),
+    (
+            {"recipient_mail": "test@mail.com"},
+            {'detail': [
+                {'loc': ['body', 'sender_name'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'},
+                {'loc': ['body', 'recipient_name'],
+                 'msg': 'field required',
+                 'type': 'value_error.missing'}
+            ]}
+    ),
+])
```

Comment:
Consider placing in a variable for readability.

Reply:
I wrote a helper function to read the error responses
now the parameters are more readable